backend/test/dataset/test_list_dataset_files.py

```diff
@@ -43,103 +46,164 @@ def build_mock_event(dataset: DatasetModel):
         {
-            "Keys": [
-                {"key": "global/datasets/example_dataset/file1.txt", "size": 643},
-                {"key": "global/datasets/example_dataset/file2.png", "size": 1243},
+            "pageSize": 1000,
+            "prefix": "global/datasets/example_dataset/",
+            "bucket": "mlspace-data-bucket",
+            "contents": [
+                {"key": "global/datasets/example_dataset/file1.txt", "size": 643, "type": "object"},
+                {"key": "global/datasets/example_dataset/file2.png", "size": 1243, "type": "object"},
+                {"prefix": "global/datasets/example_dataset/nested", "type": "prefix"},
             ],
         },
     )
 
     assert lambda_handler(build_mock_event(mock_global_dataset), mock_context) == expected_response
-
     mock_s3.list_objects_v2.assert_called_with(
-        Bucket=TEST_ENV_CONFIG["DATA_BUCKET"], Prefix="global/datasets/example_dataset/"
+        Bucket=TEST_ENV_CONFIG["DATA_BUCKET"],
+        Prefix="global/datasets/example_dataset/",
+        Delimiter="/",
     )
     mock_dataset_dao.get.assert_called_with(mock_global_dataset.scope, mock_global_dataset.name)
 
 
 @mock.patch("ml_space_lambda.dataset.lambda_functions.dataset_dao")
 @mock.patch("ml_space_lambda.dataset.lambda_functions.s3")
-def test_list_dataset_files_empty(mock_s3, mock_dataset_dao, mock_global_dataset):
+def test_list_dataset_files_success_with_prefix(mock_s3, mock_dataset_dao, mock_global_dataset):
     mock_dataset_dao.get.return_value = mock_global_dataset
     mock_s3.list_objects_v2.return_value = {
+        "MaxKeys": 1000,
+        "Prefix": "global/datasets/example_dataset/nested/",
         "Contents": [
-            {"Key": "global/datasets/example_dataset/", "Size": 0},
-        ]
+            {"Key": "global/datasets/example_dataset/nested/file3.txt", "Size": 346},
+            {"Key": "global/datasets/example_dataset/nested/file4.png", "Size": 3421},
+        ],
     }
 
     expected_response = generate_html_response(
         200,
         {
-            "Keys": [],
+            "pageSize": 1000,
+            "prefix": "global/datasets/example_dataset/nested/",
+            "bucket": "mlspace-data-bucket",
+            "contents": [
+                {"key": "global/datasets/example_dataset/nested/file3.txt", "size": 346, "type": "object"},
+                {"key": "global/datasets/example_dataset/nested/file4.png", "size": 3421, "type": "object"},
+            ],
         },
     )
 
-    assert lambda_handler(build_mock_event(mock_global_dataset), mock_context) == expected_response
-
+    assert lambda_handler(build_mock_event(mock_global_dataset, {"prefix": "nested/"}), mock_context) == expected_response
     mock_s3.list_objects_v2.assert_called_with(
-        Bucket=TEST_ENV_CONFIG["DATA_BUCKET"], Prefix="global/datasets/example_dataset/"
+        Bucket=TEST_ENV_CONFIG["DATA_BUCKET"],
+        Prefix="global/datasets/example_dataset/nested/",
+        Delimiter="/",
```

Comment:
It will always be `/`.

> - Delimiter (string) –
>
> Causes keys that contain the same string between the prefix and the first occurrence of the delimiter to be rolled up into a single result element in the CommonPrefixes collection. These rolled-up keys are not returned elsewhere in the response. Each rolled-up result counts as only one return against the MaxKeys value.
>
> **Note** Directory buckets - For directory buckets, `/` is the only supported delimiter.

https://boto3.amazonaws.com/v1/documentation/api/latest/reference/services/s3/client/list_objects_v2.html

Reply:
If it is always slash, then this should be a constant right? Why is it being attached as a property everywhere?

Reply:
Technically we're not using [directory buckets](https://docs.aws.amazon.com/AmazonS3/latest/userguide/directory-buckets-overview.html) at the moment we're using general purpose buckets. I don't think this is something we need to get lost in the weeds with though. If the hang up is us having `"/"` explicitly in multiple places we can create a single const for `DEFAULT_S3_DELIMITER` or something similar. 

It will always be `/` for us because that's how we've defined our hierarchy - technially we could rework our prefixes to be `global-datasets-<datasetname>` and then we could set the prefix to be `-` but that would be confusing and wouldn't match what people generally think of when they see a `/` in a file "key".